cli/src/cli.rs

```diff
@@ -1168,16 +1238,194 @@ fn send_and_confirm_transactions_with_spinner<T: Signers>(
     }
 }
 
+fn read_and_verify_elf(program_location: &str) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+    let mut file = File::open(program_location).map_err(|err| {
+        CliError::DynamicProgramError(format!("Unable to open program file: {}", err))
+    })?;
+    let mut program_data = Vec::new();
+    file.read_to_end(&mut program_data).map_err(|err| {
+        CliError::DynamicProgramError(format!("Unable to read program file: {}", err))
+    })?;
+
+    // Verify the program
+    Executable::<BPFError, ThisInstructionMeter>::from_elf(
+        &program_data,
+        Some(|x| bpf_verifier::check(x, false)),
+        Config::default(),
+    )
+    .map_err(|err| CliError::DynamicProgramError(format!("ELF error: {}", err)))?;
+
+    Ok(program_data)
+}
+
+fn complete_partial_init(
```

Comment:
Unfortunately the program commands are in the general `cli.rs` instead of a separate file (for now I guess).  So while the commands are in `cli.rs`, I think `complete_partial_init()` needs a more program/loader specific name.